app/services/content_embed_service.rb

```diff
@@ -0,0 +1,83 @@
+class ContentEmbedService
+  include ApplicationHelper
+
+  SUPPORTED_DOCUMENT_TYPES = %w[contact].freeze
+  UUID_REGEX = /([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})/
+  EMBED_REGEX = /({{embed:(#{SUPPORTED_DOCUMENT_TYPES.join('|')}):#{UUID_REGEX}}})/
```

Comment:
A few thoughts about the format here (some of this is bike shedding)

- `{{` vs `[` or `[[` - we'd probably want to do some user research on these formats - content designers are not used to entering curly braces currently, as markdown / govspeak doesn't use them. Although ideally people won't be typing them out by hand (visual editor / insert object menu item), if it looks like the `{{` syntax is scary / confusing we should consider alternatives. We'll have to make sure whatever we choose doesn't collide with existing syntax.
- The `embed:type:guid` syntax comes from ADR004, but it doesn't really justify why that syntax is needed / preferable. Particularly if we're using curly braces, I don't think `embed:` is needed to disambiguate from other things. I think the `type` bit is probably helpful namespacing so the users can see what type of embed it is, and I guess it makes the edition search a little faster too.
- We might want to be able to specify which field we're referring to (here we're using `title`, but you could imagine wanting to do `{{contact:government_digital_service:office_address}}` or `{{contact:government_digital_service:foi_email_address}}` (i.e. referring to specific fields on complex objects). I think it would be overcomplicating things to allow for JMESPath expressions or anything like that.
- I think the use of UUIDs will be problematic for a couple of UX reasons - firstly they're intimidating, secondly they're very hard to review in 2i. I think we should aim to use friendly IDs instead, although that will open several new cans of worms (e.g. should friendlyID be part of a document, or an edition? If an edition, can it change from one edition to the next? do friendlyIDs have to be globally unique? etc.)

All of that said, I think this pattern is fine for the purposes of this spike. Just wanted to get those thoughts down / start a discussion if people disagree on any bits.